File: src/main/java/com/baulsupp/oksocial/location/Location.java
package com.baulsupp.oksocial.location;

import java.util.Optional;

public class Location {
  public final double latitude;
  public final double longitude;

  public Location(double latitude, double longitude) {
    this.latitude = latitude;
    this.longitude = longitude;
  }

  @Override public String toString() {
    return latitude + "," + longitude;
  }

  public static Optional<Location> latLong(double latitude, double longitude) {
    return Optional.of(new Location(latitude, longitude));
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/location/Location.kt
package com.baulsupp.oksocial.location

import java.util.Optional

class Location(val latitude: Double, val longitude: Double) {

    override fun toString(): String {
        return latitude.toString() + "," + longitude
    }

    companion object {

        fun latLong(latitude: Double, longitude: Double): Optional<Location> {
            return Optional.of(Location(latitude, longitude))
        }
    }
}
